src/batch.js

```diff
@@ -1,28 +1,221 @@
 'use strict'
 
+const router = require('lambda-router/src/router')
+const { expectation } = require('sinon')
+
 const { HttpError } = require('./httpError')
 
 const batchDefaultConfig = {
-  maxBatchSize: 20
+  maxBatchSize: 20,
+  handler: undefined
 }
 
 module.exports = {
   batchHandler
 }
 
-async function batchHandler({ routes, config, extractPathParameters }, event, context) {
+async function batchHandler({ route, config }, event, context) {
   config = { ...batchDefaultConfig, ...config }
+  const body = event.body
 
   // Validate
-  // validate json body has been parsed
-  // validate body.requests has array of requests
-  // validate each body.requests[]
+  validateBatchRequest(body, config.maxBatchSize)
 
   // build graph of requests
+  const orderedRequests = tSort(body.requests)
+  const { asyncRequests, syncRequests } = splitAsyncRequests(orderedRequests)
 
   // execute requests
+  const results = await Promise.all([
+    (async () => {
+      const syncResults = []
+      for (let req of syncRequests) {
+        syncResults.push(await executeRequest(config, route, event, req))
+      }
+      return syncResults
+    })(),
+    ...asyncRequests.map(executeRequest.bind(null, config, route, event))
+  ])
 
   // return response
+  return {
+    statusCode: 200,
+    body:{ responses: results.map((result, index) => {
+      if (index === 0){
+        return result.map((result, index) => {
+          return {
+            id: syncRequests[index].id,
+            status: result.statusCode,
+            body: result.body ? JSON.parse(result.body) : undefined,
+            headers: result.headers
+          }
+        })
+      }
+      return {
+        id: asyncRequests[index - 1].id,
+        status: result.statusCode,
+        body: result.body ? JSON.parse(result.body) : undefined,
+        headers: result.headers
+      }
+    })},
+    headers:{}
+  }
 }
 
-function validateBatchRequest(body) {}
+async function executeRequest(config, route, event, request) {
+  //Parse url into pathParameters and multiValueQueryStringParameters formats
+  const urlAndQuerystring = request.url.split('?')
+  let queryStringParameters = {}
+  if (urlAndQuerystring.length > 1) {
```

Comment:
I think its worth extracting this logic into a function that can be unit-tested independently